keys/cryptostore/encoder.go

```diff
@@ -16,45 +18,53 @@ var (
 //
 // This should use a well-designed symetric encryption algorithm
 type Encoder interface {
-	Encrypt(key crypto.PrivKey, pass string) ([]byte, error)
-	Decrypt(data []byte, pass string) (crypto.PrivKey, error)
-}
-
-func secret(passphrase string) []byte {
-	// TODO: Sha256(Bcrypt(passphrase))
-	return crypto.Sha256([]byte(passphrase))
+	Encrypt(privKey crypto.PrivKey, passphrase string) (saltBytes []byte, encBytes []byte, err error)
```

Comment:
Is there a convention around it? For some reason this way felt more natural to me. :-)

If most crypto implementations follow the other order I can change it.

Reply:
don't think there is a convention out there, just returning data first feels logical to me :) when I encrypt something, I need a result in the first place. and only then salt.

bcrypt does not expose salt at all.

also, what if want to provide my own salt, like:

```
Encrypt(privKey crypto.PrivKey, passphrase string, salt []byte) (encBytes []byte, err error)
```

or we won't be needing this?

Reply:
I don't think we need to expose the ability to add a custom salt. I have a hard time seeing a usecase for that. 

Mh, I think we should stick with the current ordering of parameters, since it comes from mintkey to some extend. 

@melekes Are you giving your thumbs up for merging?